url.go

```diff
@@ -542,12 +554,12 @@ func parseIdentifierURL(url *gourl.URL) (*URL, error) {
 	// Optional
 	if r.Architecture != "" {
 		if err := ValidateArchitecture(r.Architecture); err != nil {
```

Comment:
ValidateArchitecture doesn't check IsSupportedArch, it only does a regex for strings that start with an alphabetic character and end with an alpha/numeric character. We should check IsSupported in both cases, where there is only an arch or both a series and arch, to be consistent.